src/aspire/source/image.py

```diff
@@ -98,6 +100,16 @@ def __getitem__(self, indices):
         return self.fun(indices)
 
 
+def as_copy(func):
```

Comment:
I know it was undecided whether we would keep the decorator, but I actually like it... if future developers decide immutable was one of my bad decisions one day, it will be easier to remove.

Anyway, ready to add a docstring. Just something brief about the intent is fine.

Reply:
docstring added